src/lib/resolver/functions-map.handler-resolver.ts

```diff
@@ -0,0 +1,25 @@
+
+import { MessageHandlingCollection } from "../collection/message-handling.collection";
+import { MessageTypeExtractorInterface } from "../extractor/message-type-extractor.interface";
+import { MessageHandlerResolverInterface } from "./message-handler-resolver.interface";
+
+/**
+ * Provides the ability to resolve a set of handlers bound to a given message.
+ */
+export class FunctionsMapHandlerResolver implements MessageHandlerResolverInterface {
+
+  constructor(
+      private _messageHandlingCollection: MessageHandlingCollection,
+      private _extractor: MessageTypeExtractorInterface
+  ) {}
+
+    /**
+     * @inheritDoc
+     */
+    getHandlers(message: any): Function[] {
+      // Extracts the identifier.
+      const identifier = this._extractor.extract(message);
+      // Gets the function handlers based on the message identifier.
+      return this._messageHandlingCollection.getHandlers(identifier);
```

Comment:
Are you sure the handlers in the collection are functions?
I don't see anything that forces it.